MessageProcessorModule/src/main/java/com/lotus/iot/mpm/Application.java

```diff
@@ -0,0 +1,150 @@
+package com.lotus.iot.mpm;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.lotus.iot.mpm.api.objectmodel.request.SubscribeRequest;
+import com.lotus.iot.mpm.api.objectmodel.response.AllAssetsInStateResponse;
+import com.lotus.iot.mpm.api.objectmodel.response.AssetCurrentStatusResponse;
+import com.lotus.iot.mpm.api.objectmodel.response.GeneralResponse;
+import com.lotus.iot.mpm.eventhandler.AssetStatusChangeQueue;
+import com.lotus.iot.mpm.eventhandler.EventHandlerFacade;
+import com.lotus.iot.mpm.eventhandler.objectmodel.AssetStatusEvent;
+import com.lotus.iot.mpm.eventhandler.objectmodel.SampleListener;
+import com.lotus.iot.mpm.handler.MessageHandler;
+import com.lotus.iot.mpm.objectmodel.AssetStatusMessage;
+import com.lotus.iot.mpm.objectmodel.type.StatusType;
+import com.lotus.iot.mpm.report.ReportHandler;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Niloufar Mazloumpour
+ * @mail niloufar@mazloumpour.net
+ * @since 2018-05-22
+ */
+@SpringBootApplication
+public class Application {
+    public static void main(String[] args) {
+        ConfigurableApplicationContext applicationContext = SpringApplication.run(Application.class);
+
+        subscribeSomeListeners(applicationContext);
+
+        if (args.length > 0) {
+            File file = new File(args[0]);
+            sendMessages(file, applicationContext);
+        }
+
+        waitForProcessToFinish(applicationContext);
```

Comment:
I added this waiting time so the application will finish all its process before I start to get reports. It is not necessary for the outcome but I wanted the console output to be clear and readable. First program will do all the process and print notifications in the console. Then it will print the reports.
I could have also divide output to separate files for this matter.